lib/validators/checklist_result_extractor.rb

```diff
@@ -4,102 +4,96 @@ module ChecklistResultExtractor
 
     # find all nodes that fulfill the data criteria
     def find_dc_node(template, checked_criteria, source_criteria)
+      @checked_criteria = checked_criteria
+      @source_criteria = source_criteria
+      @template = template
       passing = false
       # find nodes to search for the criteria
-      reason_template, nodes = template_nodes(source_criteria, checked_criteria)
+      reason_template, nodes = template_nodes
       # if the checked criteria has a code, the code and attributes will be checked
-      # if the checked criteria does not have a code (e.g. Transfers), on the attiributes will be checked
-      if checked_criteria.code || checked_criteria.negated_valueset
-        neg_vs = checked_criteria.negated_valueset ? checked_criteria['selected_negated_valueset'] : nil
-        codenodes = find_template_with_code(nodes, template, checked_criteria, reason_template, neg_vs)
+      if @checked_criteria.code || @checked_criteria.negated_valueset
+        neg_vs = @checked_criteria.negated_valueset ? @checked_criteria['selected_negated_valueset'] : nil
+        codenodes = find_template_with_code(nodes, reason_template, neg_vs)
         # When you get nodes that include a code, determine if it meets additinal criteria
-        passing = passing_dc?(codenodes, source_criteria, checked_criteria)
-      elsif checked_criteria.attribute_code # CMS188v6
-        codenodes = find_template_with_attribute([@file], template, checked_criteria.attribute_code)
-        passing = true unless codenodes.empty?
+        passing = passing_dc?(codenodes)
       end
       if passing
-        checked_criteria.passed_qrda = true
-        checked_criteria.save
+        @checked_criteria.passed_qrda = true
+        @checked_criteria.save
       end
     end
 
-    def passing_dc?(codenodes, source_criteria, checked_criteria)
```

Comment:
This looks like a paradigm change to expecting the internal state to track the current source_criteria, checked_criteria. I think that is okay because this and below methods are used only by find_dc_node, which establishes the state. But if we have the expectation that all of these methods will only ever be called internally, we should mark them as private.